src/main/java/NestedNullRefactoring.java

```diff
@@ -0,0 +1,155 @@
+import org.eclipse.jdt.core.dom.IMethodBinding;
+import org.eclipse.jdt.core.dom.InfixExpression;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+import org.eclipse.jdt.core.dom.MethodInvocation;
+import org.eclipse.jdt.core.dom.NullLiteral;
+import org.eclipse.jdt.core.dom.ParenthesizedExpression;
+import org.eclipse.jdt.core.dom.PrefixExpression;
+import org.eclipse.jdt.core.dom.PrimitiveType;
+import org.eclipse.jdt.core.dom.ReturnStatement;
+import org.eclipse.jdt.core.dom.SimpleName;
+import org.eclipse.jdt.core.dom.Statement;
+import org.eclipse.jdt.core.dom.Type;
+import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
+
+/**
+ * This class represents a refactoring in which invocations to private one line
+ * methods which return the nullness of a value are replaced with the method's
+ * one-line null check. Preserves semantics by copying the invoked code
+ * precisely.
+ */
+public class NestedNullRefactoring extends Refactoring {
+	public static final String NAME = "NestedNullRefactoring";
+
+	private final Dictionary<IMethodBinding, Expression> applicableMethods;
+
+	public NestedNullRefactoring() {
+		applicableMethods = new Hashtable<>();
+	}
+
+	@Override
+	public boolean isApplicable(ASTNode node) {
+		if (node instanceof MethodInvocation invocation) {
+			return isApplicable(invocation);
+		}
+
+		if (node instanceof MethodDeclaration declaration) {
+			return isApplicable(declaration);
+		}
+
+		return false;
+	}
+
+	/*
+	 * Returns true iff the provided invocation is of a registered one-line method
+	 * that returns the result of a null check
+	 */
+	public boolean isApplicable(MethodInvocation invocation) {
+		if (applicableMethods.get(invocation.resolveMethodBinding()) != null) {
+			System.out.println("[DEBUG] Invocation of applicable method found");
+			return true;
+		}
+		return false;
+	}
+
+	/*
+	 * Returns true iff Node is a one-line private or package-private method that
+	 * returns the result of a null check
+	 */
+	public boolean isApplicable(MethodDeclaration declaration) {
+		Type retType = declaration.getReturnType2();
+		boolean returnsBoolean = (retType.isPrimitiveType()
+				&& ((PrimitiveType) retType).getPrimitiveTypeCode() == PrimitiveType.BOOLEAN);
+		if (!(returnsBoolean)) {
+			return false;
+		}
+
+		// Ensure the method declaration is private
+		if (!((declaration.getModifiers() & Modifier.PRIVATE) == Modifier.PRIVATE)) {
+			return false;
+		}
+
+		// Checks if there are any parameters
+		// TODO: Make work with parameters in case of methods that take variables to
+		// check as inputs, such as `checkObject(Object o) { return o !=null }}`
+		boolean hasParams = declaration.parameters().size() > 0;
+		if (hasParams) {
+			return false;
+		}
+
+		Block body = declaration.getBody();
+		List<Statement> stmts = body.statements();
+
+		boolean isOneLine = stmts.size() == 1;
+		if (!isOneLine) {
+			return false;
+		}
+
+		// Possible unneccessary since we already confirmed return type is not void
```

Comment:
This is important, because it helps make sure that this code won't trigger for other methods that have other kinds of behavior.

Reply:
e.g., that have side-effects. I think we can remove this comment.